R/model.R

```diff
@@ -225,17 +236,47 @@ compile_method <- function(quiet = TRUE,
                            force_recompile = FALSE,
                            #deprecated
                            threads = FALSE) {
-  if (all(nzchar(self$exe_file()))) {
-    exe <- cmdstan_ext(strip_ext(self$stan_file()))
-  } else {
-    exe <- cmdstan_ext(strip_ext(self$exe_file()))
+  if (length(cpp_options) == 0) {
+    if (!is.null(private$cpp_options_)) {
```

Comment:
@rok-cesnovar If `private$cpp_options_` doesn't have anything then, the way you've coded the CmdStanModel object, it will be an empty `list()` rather than `NULL`. So `!is.null()` will always return TRUE here. I'll push a quick commit to fix this.